packages/p2p/src/components/modal-manager/modals/buy-sell-modal/buy-sell-modal.tsx

```diff
@@ -0,0 +1,169 @@
+import { observer } from '@deriv/stores';
+import { useModalManagerContext } from 'Components/modal-manager/modal-manager-context';
+import AddPaymentMethodForm from 'Pages/my-profile/payment-methods/add-payment-method/add-payment-method-form.jsx';
+import BuySellForm from 'Pages/buy-sell/buy-sell-form.jsx';
+import BuySellFormReceiveAmount from 'Pages/buy-sell/buy-sell-form-receive-amount.jsx';
+import { useStores } from 'Stores';
+import BuySellModalFooter from './buy-sell-modal-footer';
+import BuySellModalTitle from './buy-sell-modal-title';
+import BuySellModalError from './buy-sell-modal-error';
+
+const BuySellModal = () => {
+    const { hideModal, is_modal_open, showModal } = useModalManagerContext();
+    const { buy_sell_store, floating_rate_store, general_store, my_profile_store, order_store } = useStores();
+    const { is_buy, selected_ad_state } = buy_sell_store;
+    const { balance } = general_store;
+    const { should_show_add_payment_method_form } = my_profile_store;
+
+    const history = useHistory();
+    const location = useLocation();
+    const [error_message, setErrorMessage] = React.useState('');
+    const [is_submit_disabled, setIsSubmitDisabled] = React.useState(false);
+    const [is_account_balance_low, setIsAccountBalanceLow] = React.useState(false);
+    const submitForm = React.useRef(() => {
+        // do nothing
+    });
+
+    const show_low_balance_message = !is_buy && is_account_balance_low;
+
+    const setSubmitForm = (submitFormFn: () => void) => (submitForm.current = submitFormFn);
+
+    const onCancel = () => {
+        if (should_show_add_payment_method_form) {
+            if (general_store.is_form_modified) {
+                showModal({
+                    key: 'CancelAddPaymentMethodModal',
+                    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+                    // @ts-ignore TODO: fix typings in CancelAddPaymentMethodModal and make them optional and remove this comment
+                    props: {},
+                });
+            } else {
+                my_profile_store.hideAddPaymentMethodForm();
+            }
+        } else {
+            hideModal();
+            buy_sell_store.fetchAdvertiserAdverts();
+        }
+        floating_rate_store.setIsMarketRateChanged(false);
+    };
+
+    const onConfirmClick = (order_info: { id: string }) => {
+        const current_query_params = new URLSearchParams(location.search);
+        current_query_params.append('order', order_info.id);
+        general_store.redirectTo('orders', { nav: { location: 'buy_sell' } });
+        history.replace({
+            pathname: routes.p2p_orders,
+            search: current_query_params.toString(),
+            hash: location.hash,
+        });
+        order_store.setOrderId(order_info.id);
+        hideModal();
+        buy_sell_store.fetchAdvertiserAdverts();
+        buy_sell_store.setShowAdvertiserPage(false);
+    };
+
+    React.useEffect(() => {
```

Comment:
Is this reaction is no longer valid or it being moved somewhere?

![image](https://github.com/farrah-deriv/deriv-app/assets/125247833/d5d7f8b2-569d-45e9-9d34-439f46d5e3c3)

Reply:
nono so its still needed for when there is any change to the market rate set by the user, the reason why i moved it to the store as separating reactions from components and moving them to stores has several benefits. Firstly, it separates concerns, allowing you to keep the concerns of rendering and user interaction separate from managing state and reacting to changes. Secondly, it promotes reusability, as other components can access the same store and benefit from the reactive behavior defined by the reactions (i did this for the advertiser page's reaction in my-profile refactor since there was another page that had the same reaction). Thirdly, it enables centralized state management, making it easier to track and understand how different parts of the application interact and respond to state changes. Finally, it simplifies testing, as you can write unit tests specifically targeting the reactions without needing to mount or render the entire component. 👍

Reply:
its a mouthful but, other packages also do this but i talked with farrah and niloofar about removing these reactions into the stores 👍

Reply:
Thanks for the explanation 😄 , but there are 2 reactions and i can only see that you moved one reaction but not that one 👆

Reply:
i dont need to check for the previous buy sell advert and compare it with a new one since the error for the rate has changed will come from BE hence why i deleted it

Reply:
ah wait sorry for the confusion, what i mean is we do need to check for the previous and new advert and there wasn't the need to check for the error code since this is where we can check if the rate has changed

Reply:
OHHH alright, now i understand why you remove it 🙇

Reply:
yeahhh sorry i got confused looking at the above reaction you screenshotted thinking i left that one 😆 but yeah hahaha

Reply:
@ameerul-deriv can we create a hook for it instead?

Reply:
hmmm if we make a hook, I don't really see how it can track the change of the adverts properties since the reaction will help take care of that unless it was subscribed to and we regularly get the change. In this case the advert is set by getting the advert within form_props which is set in the buy sell form so there would be a lot that needs to be changed if we try to create a hook